src/machines/sketchSolveMode.ts

```diff
@@ -0,0 +1,136 @@
+import { createMachine, sendTo, setup, type AnyActorLogic } from 'xstate'
+import { modelingMachineDefaultContext } from '@src/machines/modelingMachine'
+import type {
+  ModelingMachineContext,
+  SetSelections,
+  MouseState,
+  SegmentOverlayPayload,
+} from '@src/machines/modelingMachine'
+import type { PathToNode } from '@src/lang/wasm'
+import { machine as centerRectToolMachine } from '@src/machines/centerRectTool'
+import { machine as dimensionToolMachine } from '@src/machines/dimensionTool'
+
+type EquipTool = 'dimension' | 'center rectangle'
+
+// Tool registry for dynamic tool selection (including fallback)
+const toolRegistry: Record<EquipTool, any> = {
+  'center rectangle': centerRectToolMachine,
+  dimension: dimensionToolMachine,
+}
+
+// Helper function to get tool machine
+const getToolMachine = (tool: EquipTool) =>
+  toolRegistry[tool] || centerRectToolMachine
+
+export type SketchSolveMachineEvent =
+  | { type: 'exit' }
+  | { type: 'update selection'; data?: SetSelections }
+  | { type: 'unequip tool' }
+  | { type: 'equip tool'; data: { tool: EquipTool } }
+  | { type: 'add point'; data: [x: number, y: number] }
+  | { type: 'Set mouse state'; data: MouseState }
+  | { type: 'Set Segment Overlays'; data: SegmentOverlayPayload }
+  | { type: 'Delete segment'; data: PathToNode }
+  | { type: 'change tool'; data: { tool: EquipTool } }
+  | { type: 'Center camera on selection' }
+  | { type: 'click in scene'; data: [x: number, y: number] }
+
+export const sketchSolveMachine = setup({
+  types: {
+    context: {} as ModelingMachineContext,
+    events: {} as SketchSolveMachineEvent,
+  },
+  actions: {
+    'send unequip to tool': sendTo('toolInvoker', {
+      type: 'unequip',
+    }),
+    'send update selection to equipped tool': sendTo('toolInvoker', {
+      type: 'update selection',
+    }),
+    'send updated selection to move tool': sendTo('moveTool', {
+      type: 'update selection',
+    }),
+    'request unequip': sendTo('toolInvoker', {
+      type: 'unequip',
+    }),
+  },
+  actors: {
+    moveToolActor: createMachine({
+      /* ... */
+    }),
+    toolInvoker: (({ input }: { input: { tool: EquipTool } }) =>
+      getToolMachine(input.tool)) as unknown as AnyActorLogic,
```

Comment:
I fear this type coercion is covering up something that I don't understand about how to procedurally invoke an actor like this. If I remove the coercions and change the type on the `toolRegistry` to be `Record<ToolEquip, UnknownActorLogic> it still doesn't work for me, with the following error:
```
1. Type '({ input }: { input: { tool: EquipTool; }; }) => UnknownActorLogic' is not assignable to type 'UnknownActorLogic'. [2322]
```

Which is very odd: it seems that there needs to be top-level wrapper function of some sort, kind of like the `fromPromise` we've used elsewhere.

Reply:
We've agreed that this is a problem we can overcome as we implement each tool in its turn